lib/mongo/session.rb

```diff
@@ -770,7 +779,7 @@ def add_txn_num!(command)
     def add_txn_opts!(command, read)
       command.tap do |c|
         # The read preference should be added for all read operations.
-        if read && txn_read_pref = txn_read_preference
+        if false && read && txn_read_pref = txn_read_preference
```

Comment:
This statement should never evaluate to true, correct?

Reply:
Right,  I forgot to remove this piece.